### DISCUSSION TITLE
Consider a call for maintainers

### DISCUSSION THREAD

**Discussion by mrlife** [snapappointments/bootstrap-select#2614] [General]
Open source software is a boon for software developers around the world. It also takes a lot of effort to maintain a repository. Please consider recruiting some maintainers to help keep the ball rolling.

e.g.  https://github.com/selectize/selectize.js/issues/752

cc @caseyjhol @brockholzer

**Comment by NicolasCARPi** [comment]
@caseyjhol I would be willing to help with issue triage (and closing the old ones) and pr review.

**Comment by caseyjhol** [comment]
@NicolasCARPi That's great - thank you! That's definitely what I need the most help with currently. Here is a pre-written message you can use on bug reports that are missing requirements. I'll try to get you access to close/label tickets soon. In the meantime you can just tag me in a comment if you think it should be closed.

Bug reports must include a live demo of the problem. Per our [contributing guidelines](https://github.com/snapappointments/bootstrap-select/blob/master/CONTRIBUTING.md), please create a reduced test case (you can use our preconfigured [Plunker](https://developer.snapappointments.com/bootstrap-select/playground/)), and report back with:

- your link
- Bootstrap version
- bootstrap-select version
- specific browser and OS details.

If you're not using the latest version of bootstrap-select, please try that as well.

**Comment by NicolasCARPi** [comment]
@caseyjhol I'm glad you accept my proposition. I think I'll start by modifying the issue template to include the required details so at least new issues have that, it seems to be a generic template as of now. I'll then gradually go over the existing issues with the close hammer at hand.

I think for a repo of this size we can aim for 100 open issues and 10 open PR. PR should be merged or closed after 30 days max ideally.

Cheers,
~Nico

**Comment by caseyjhol** [comment]
@NicolasCARPi The bug issue template is customized for bootstrap-select currently, but many people choose to ignore it anyway. It could definitely be improved, though. I think perhaps short and sweet would be better to ensure people actually read through it.

**Comment by caseyjhol** [comment]
@NicolasCARPi Also, before merging in any PRs you think look good, please flag me first so I can review. Thanks!

**Comment by NicolasCARPi** [comment]
Well, this is weird, I got the invitation email, get on the invitation page, click Accept, first I get an error that my browser did something wrong, second time I get a 404, and now it tells me Invitation is invalid....  ¯\_(ツ)_/¯

I'll try later...

edit: opened a support ticket for it. Maybe I'll need to decline and you'll need to invite again.

**Comment by NicolasCARPi** [comment]
@caseyjhol I have declined the invitation (as it is not working), could you kindly make a new one (so I stop pinging you and can actually close stuff myself)?

Cheers,
~Nico

**Comment by NicolasCARPi** [comment]
@caseyjhol Thanks for the invitation, this time it worked. But it seems it is only "view" rights, which doesn't mean a lot for a public repository. The idea was that I would have rights to label and close issues, so I can be an effective help, do you think you can extend rights like so?

**Reply by caseyjhol** [reply]
Yes, I tried giving you "triage" access previously and ran into issues with the invitation, so I tried just "Read" access this time to see if I could upgrade you to triage afterwards. I'm getting a "Something went wrong" error when trying to do so, however. Checking into it.

**Reply by caseyjhol** [reply]
@NicolasCARPi You now have access to triage issues.

**Reply by NicolasCARPi** [reply]
can confirm, thanks Casey! :)

**Comment by NicolasCARPi** [comment]
Hello @caseyjhol, it seems you're not active these days on the repo. Would you be ok if I start merging simple PRs? Please let us know what's your current state of mind regarding this project, I know being an open source maintainer is a big task, and it's fine if you wish to take a break from the constant flow of issues. We're here to help ;)

**Comment by mrlife** [comment]
@caseyjhol @NicolasCARPi Could you use more help? There are 50 open PRs and over 200 open issues. The last commit was a couple months ago. Many have been waiting for the official release of 1.14, etc.

This is a great product without a great competitor. Please keep it going!

**Reply by NicolasCARPi** [reply]
@mrlife Casey just merged two PR that were tagged as mergeable by me, so he's still around, but yeah it doesn't look like he's very active and I asked some questions that are still unanswered (like https://github.com/snapappointments/bootstrap-select/issues/2617). So while I was filled with good will as the start, I must say that not having answers to my interrogations kinda stopped me in my tracks on helping out this project. I still want to participate as I use this in my main project but @caseyjhol you need to please clarify your position. Are you still working on this project? Are you fed up with it? (we would understand this! It can be taxing to maintain such a popular project)

Basically, can I take over and start merging stuff I think is good and make releases or should I just help with issue and PR triage? Also, can the `master` branch be deleted? (something I asked elsewhere)

@mrlife your help is of course very welcome :) You don't need write access, just comment "this can be closed / this is a dupe / this can be merged" and I'll do what is necessary. I've already closed A LOT of issues but there are always more ;)
